builtin/providers/aws/resource_aws_lambda_function.go

```diff
@@ -487,3 +541,18 @@ func validateVPCConfig(v interface{}) (map[string]interface{}, error) {
 
 	return config, nil
 }
+
+func validateEnvironment(v interface{}) (map[string]interface{}, error) {
```

Comment:
@kwilczynski Well at the moment yes! I made the assumption that as Environment is an object, it may contain more than the attribute `variables` in the future. This method would allow us to validate the entire structure.
It also avoids me to duplicate the code in the creation and when environment has changes. 
Would you suggest something else?